module/data/item/templates/activities.mjs

```diff
@@ -134,4 +204,133 @@ export default class ActivitiesTemplate extends SystemDataModel {
   prepareFinalActivityData(rollData) {
     UsesField.prepareData.call(this, rollData);
   }
+
+  /* -------------------------------------------- */
+  /*  Helpers                                     */
+  /* -------------------------------------------- */
+
+  /**
+   * Retrieve information on available uses for display.
+   * @returns {{value: number, max: number, name: string}}
+   */
+  getUsesData() {
+    return { value: this.uses.value, max: this.uses.max, name: "system.uses.value" };
+  }
+
+  /* -------------------------------------------- */
+  /*  Shims                                       */
+  /* -------------------------------------------- */
+
+  /**
+   * Apply shims for data removed from ActionTemplate & ActivatedEffectTemplate.
+   * @this {ItemDataModel}
+   */
+  static _applyActivityShims() {
+    const shim = (template, property, get) => {
+      if ( property in this ) return;
+      Object.defineProperty(this, property, {
+        get: () => {
+          foundry.utils.logCompatibilityWarning(
+            `The \`${property}\` property on \`${template}\` has been deprecated.`,
+            { since: "DnD5e 4.0", until: "DnD5e 4.4", once: true }
+          );
+          return get();
+        },
+        configurable: true,
+        enumerable: false
+      });
+    };
+    const addShims = (template, shims) => Object.entries(shims).forEach(([key, method]) => shim(template, key, method));
+    const firstActivity = this.activities.contents[0] ?? {};
+
+    addShims("ActionTemplate", {
+      ability: () => firstActivity.ability ?? null,
+      actionType: () => firstActivity.actionType ?? "",
+      attack: () => {
+        const activity = this.activities.getByType("attack")[0] ?? {};
+        return {
+          bonus: activity.attack?.bonus ?? "",
+          flat: activity.attack?.flat ?? false
+        };
+      },
+      chatFlavor: () => firstActivity.description?.chatFlavor ?? "",
+      critical: () => {
+        const activity = this.activities.getByType("attack")[0] || this.activities.getByType("damage")[0];
+        return {
+          threshold: activity?.attack?.critical?.threshold ?? null,
+          damage: activity?.damage?.critical?.bonus ?? ""
+        };
+      },
+      dammage: () => ({ parts: [], versatile: "" }),
```

Comment:
There's a type, but also, should this not be pulling from the attack, damage, or save activities?
```suggestion
      damage: () => ({ parts: [], versatile: "" }),
```